src/elm/Page/Community/Settings/Currency.elm

```diff
@@ -0,0 +1,560 @@
+
+                        _ ->
+                            UR.logImpossible msg [ "WithoutCommunity" ]
+                   )
+
+        GotSubmitResponse (Err val) ->
+            { model | isLoading = False }
+                |> UR.init
+                |> UR.addExt (LoggedIn.ShowFeedback Feedback.Failure (shared.translators.t "community.error_saving"))
+                |> UR.logDebugValue msg val
+
+        CompletedLoadCommunity community ->
+            { model
+                | inviterReward = String.fromFloat community.inviterReward
+                , invitedReward = String.fromFloat community.invitedReward
+                , minimumBalance =
+                    Maybe.map String.fromFloat community.minBalance
+                        |> Maybe.withDefault "0"
+                , isLoading = False
+            }
+                |> UR.init
+
+
+withSymbolValidation : (Eos.Symbol -> Model -> Result String a) -> Field -> LoggedIn.Model -> Model -> Model
+withSymbolValidation fn field loggedIn_ model_ =
+    case loggedIn_.selectedCommunity of
+        RemoteData.Success community ->
+            fn community.symbol model_
+                |> (\result -> setErrors field result model_)
+
+        _ ->
+            model_
+
+
+savePort : ValidModel -> LoggedIn.Model -> Community.Model -> Ports.JavascriptOutModel Msg
+savePort validModel loggedIn community =
+    let
+        authorization =
+            { actor = loggedIn.accountName
+            , permissionName = Eos.samplePermission
+            }
+
+        asset amount =
+            { amount = amount
+            , symbol = community.symbol
+            }
+    in
+    { responseAddress = ClickedSubmit
+    , responseData = encodeValidModel validModel
+    , data =
+        Eos.encodeTransaction
+            [ { accountName = loggedIn.shared.contracts.community
+              , name = "update"
+              , authorization = authorization
+              , data =
+                    { asset = asset 0
+                    , logo = community.logo
+                    , name = community.name
+                    , description = community.description
+                    , inviterReward = asset validModel.inviterReward
+                    , invitedReward = asset validModel.invitedReward
+                    , hasObjectives = Eos.boolToEosBool community.hasObjectives
+                    , hasShop = Eos.boolToEosBool community.hasShop
+                    }
+                        |> Community.encodeUpdateData
+              }
+            , { accountName = loggedIn.shared.contracts.token
+              , name = "update"
+              , authorization = authorization
+              , data =
+                    { maxSupply = asset 21000000.0
+                    , minBalance = asset validModel.minimumBalance
+                    }
+                        |> Community.encodeUpdateTokenData
+              }
```

Comment:
this is the only call needed, the other one, on the `shared.contracts.community` is unnecessary on this screen

Reply:
What about updating `inviterReward` and `invitedReward` though?

Reply:
those are community settings, not token settings... Maybe all of this confusion and discussion is because Figma is wrong? 

because if the criteria for adding invitations to the token is because it envolves some type of reward, then objectives should also be there.... 

The community screen should include the `inviterReward` and `invitationReward` those are related to stuff that happens on the `network` of members, tokens can still exist outside of it and need to be like that in order for it to be compatible with other tools (like wallets).

Reply:
> those are community settings, not token settings

From our point of view, yes, I agree. However, from the user's point of view, they probably don't even know there is a token separate from the community - it's all the same thing. For them, a community has a name, a description, a currency, a logo, etc.

So the user's separation of concerns is very different from ours. We know there are two entities - a community and a token - but they don't necessarily know that (AFAIK). The way it is now, we have two screens:
- "Community Information": this is used to manage what the community "looks" like - name, logo, description, etc.
- "Currency and rewards": this is used to manage how the community works - invited reward, inviter reward and minimum balance (following this, maybe the auto invites should be in this screen as well? But IMO it doesn't really fell right in this screen)

Also, moving these two fields to "Community Information" would lead to a pretty big form there, and the "Currency and rewards" screen would have a form with a single input field. Maybe we can talk to the design team about this? @heltonlr

Reply:
> So the user's separation of concerns is very different from ours. 

That's exactly the problem, I don't see why it should be that way.

> Also, moving these two fields to "Community Information" would lead to a pretty big form there, and the "Currency and rewards" screen would have a form with a single input field. 

Not true my bro, I urge you to take a look at the [token contract](https://local.bloks.io/account/cambiatus.tk?nodeUrl=http%3A%2F%2Fstaging.cambiatus.io&coreSymbol=SYS&systemDomain=eosio&loadContract=true&tab=Tables&account=cambiatus.tk&scope=cambiatus.tk&limit=100&table=expiryopts)!

A list of information related to Token:

- Current Supply (informative only)
- Max possible supply (editable)
- Min balance (editable)
- Issuer (editable -- but I think we shouldn't allow it for now)
- type (can be expiry or mcc)

Tokens can have multiple types, you are already familiar with configurations for the `mcc` one -- mutual credit clearing. But the token can be of type `expiry` that will come with more fields:

- Natural person expiration period in seconds (editable)
- Juridical person expiration period in seconds(editable)
- Renovation amount (editable)
- currency (the SYMBOL used -- view only)

So that's 10 fields. I won't count but it surely have a lot. Well this conversion is interesting because it shows how little we use from the structure that we already have. The `/community/new` is super simple and very old (you must have noticed from the code) and that shows how much we can improve there! Maybe the designs are not ideal not even considering that it has so much stuff.

I keep my position of having two screens, and not creating an different abstraction for the user, keep the one in the contract! that initial screen that kept both together was from a time that we didn't even had the `expiry` token type and a lot has changed!

Reply:
> Not true my bro, I urge you to take a look at the token contract!

Oh, I had no idea we had so many possible fields for the token!

In that case, I guess we should rename from "Currency and rewards" to something more related to token?

> you are already familiar with configurations for the mcc one -- mutual credit clearing.

Just to be sure: `mcc` is how most (all?) communities work right now, and doesn't introduce any other fields, right?

> The /community/new is super simple and very old (you must have noticed from the code) and that shows how much we can improve there!

Indeed. I think we should tackle that in another issue though - I created #514 for this

Reply:
> In that case, I guess we should rename from "Currency and rewards" to something more related to token?

Yeah I do think this screen needs improvements, it doesn't make sense that we update the whole screen to work with 
Cambiatus' features from one year ago!

> Just to be sure: mcc is how most (all?) communities work right now, and doesn't introduce any other fields, right?
> 
> I wonder if we can make these changes right away without breaking anything on other parts of the frontend or the backend (I suppose we can, but just want to be sure). Also, can a community freely change between mcc and expiry whenever they want or is there some sort of restriction around that?

There is only one community that uses `expiry`, Ticolones (which is not active). A community will not be able to change between token strategies once it was created. Maybe we can support something like this in the future but the contract doesn't allow that now

Reply:
@rafachadud can you take a look at this as well? We're moving the "inviter reward" and "invited reward" fields to the "Community information" page, and adding some fields to the "Currency and rewards" page as [listed above](https://github.com/cambiatus/frontend/pull/494#discussion_r617734936). Also, the "Currency and rewards" page should probably be renamed to something like "Token information" (you can read the reasoning behind these changes in this thread)